aea/cli/upgrade.py

```diff
@@ -0,0 +1,203 @@
+    """
+    Upgrade an item.
+
+    :param ctx: Context object.
+    :param item_type: the item type.
+    :param item_public_id: the item public id.
+    :return: None
+    """
+    agent_name = cast(str, ctx.agent_config.agent_name)
+    if not _check_upgrade_is_required(ctx, item_type, item_public_id):
+        raise click.ClickException(
+            f"Item {item_type} {item_public_id.author}/{item_public_id.name} can not be updated. Aborting..."
+        )
+
+    click.echo(
+        "Upgrading {} to '{}' for the agent '{}'...".format(
+            item_type, item_public_id, agent_name
+        )
+    )
+
+    remove_item(ctx, item_type, item_public_id)
+    add_item(ctx, item_type, item_public_id)
+    click.echo("{} for the agent '{}' upgraded".format(item_public_id, agent_name))
+
+
+def _check_upgrade_is_required(ctx: Context, item_type: str, item_public_id: PublicId):
+    """
+    Check item can be upgraded or not.
+
+    :param ctx: Context object.
+    :param item_type: the item type.
+    :param item_public_id: the item public id.
+    :return: None
+    """
+    registered_item_ids = get_items(ctx.agent_config, item_type)
+
+    def is_item_with_version_present(public_id: PublicId) -> bool:
+        if public_id.version == "latest":
+            return False
+        return public_id in registered_item_ids
+
+    if not is_item_present(ctx, item_type, item_public_id):
+        raise click.ClickException(
+            "Error: A {} with id '{}/{}' is not registered. Please use `add` command. Aborting...".format(
+                item_type, item_public_id.author, item_public_id.name
+            ),
+        )
+
+    if is_item_with_version_present(item_public_id):
+        click.echo(
+            "A {} with id '{}/{}' already has version {}. Nothing to upgrade.".format(
+                item_type,
+                item_public_id.author,
+                item_public_id.name,
+                item_public_id.version,
+            ),
+        )
+        return False
+
+    current_item_public_id = get_item_public_id_by_author_name(
+        ctx.agent_config, item_type, item_public_id.author, item_public_id.name
+    )
+
+    if (
+        current_item_public_id is None
+    ):  # pragma: nocover. already checked above, need for mypy only
+        raise click.ClickException(f"Item {item_public_id} is not registered!")
+
+    if item_public_id.version == "latest":
+        item_public_id = _get_latest_version(ctx, item_type, item_public_id)
+
+        if is_item_with_version_present(item_public_id):
+            click.echo(
+                "A {} with id '{}/{}' already has the latest version {}. Nothing to upgrade.".format(
+                    item_type,
+                    item_public_id.author,
+                    item_public_id.name,
+                    item_public_id.version,
+                )
+            )
+            return False
+
+    return True
+
+
+def _get_latest_version(
```

Comment:
I suspect this already exists somewhere..if not, it should be moved to utils

Reply:
it's done with package installation, not assumed to provide a version